src/allotropy/allotrope/schema_parser/backup_manager.py

```diff
@@ -0,0 +1,62 @@
+from collections.abc import Iterator
+from contextlib import contextmanager
+from pathlib import Path
+import shutil
+from typing import Optional, Union
+
+PathType = Union[Path, str]
+
+
+def files_equal(path1: PathType, path2: PathType) -> bool:
+    with open(str(path1)) as file1, open(str(path2)) as file2:
+        for line1, line2 in zip(file1, file2):
+            if line1 != line2 and not line1.startswith("#   timestamp:"):
+                return False
+    return True
+
+
+def _get_backup_path(path: PathType) -> Path:
+    _path = Path(path)
+    return Path(_path.parent, f".{_path.stem}.bak{_path.suffix}")
+
+
+def is_file_changed(path: PathType) -> bool:
+    if _get_backup_path(path).exists():
+        return not files_equal(path, _get_backup_path(path))
```

Comment:
nit: could extract `_get_backup_path(path)` to local var, here and in `restore_backup`